internal/noderesourcetopology/noderesourcestopology.go

```diff
@@ -0,0 +1,87 @@
+package noderesourcetopology
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nrtv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+	nrtv1alpha2attr "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2/helper/attribute"
+	"github.com/k8stopologyawareschedwg/podfingerprint"
+)
+
+type PFPCount struct {
+	Value string
+	Count int
+}
+
+func (pc PFPCount) String() string {
+	return fmt.Sprintf("PFP=%s count=%d", pc.Value, pc.Count)
+}
+
+// map nodeName -> PFPCount
+type PFPState map[string]PFPCount
+
+func (ps PFPState) Len() int {
+	return len(ps)
+}
+
+func (ps PFPState) CountReady(threshold int) int {
+	ready := 0
+	for _, count := range ps {
+		if count.Count >= threshold {
+			ready++
+		}
+	}
+	return ready
+}
+
+func (ps PFPState) AllReady(threshold int) bool {
+	for _, count := range ps {
+		if count.Count < threshold {
+			return false
+		}
+	}
+	return true
+}
+
+func (ps PFPState) Observe(nodeName, pfpValue string) int {
+	count, ok := ps[nodeName]
+	if !ok || count.Value != pfpValue {
+		ps[nodeName] = PFPCount{
+			Value: pfpValue,
+			Count: 1,
+		}
+		return 1
+	}
+	count.Count++
+	ps[nodeName] = count
+	return count.Count
+}
+
+func (ps PFPState) String() string {
+	return "TODO"
+}
+
+func GetCurrentPFPState(cli client.Client, ctx context.Context) (PFPState, error) {
+	//get nrt list and iterate over the nrts while storing the pfp value
+	var pfpState PFPState
+	nrtList := nrtv1alpha2.NodeResourceTopologyList{}
+
+	err := cli.List(ctx, &nrtList)
+	if err != nil {
+		return nil, err
+	}
+
+	for idx := range nrtList.Items {
+		nrt := &nrtList.Items[idx]
+		attr, ok := nrtv1alpha2attr.Get(nrt.Attributes, podfingerprint.Attribute)
+		if !ok {
+			return nil, fmt.Errorf("NRT %s missing PFP attribute", nrt.Name)
+		}
+		pfpState.Observe(nrt.Name, attr.Value)
+	}
+
+	return pfpState, nil
+}
```

Comment:
I'd actually extract more code from `ForNodeResourceTopologiesSettled`, all the code which gets the PFP state (including the optional `NodeName` filter) and reimplement `ForNodeResourceTopologiesSettled` on top of this helper

Reply:
Thanks for your note. no PFP updates are done thus no longer relevant.